src/thread/scheduler.zig

```diff
@@ -1,31 +1,67 @@
 const std = @import("std");
 const os = @import("root").os;
 
-const task_alloc = os.memory.vmm.backed(.Ephemeral);
+pub fn yield() void {
+  os.platform.yield(true);
+}
 
-// Creating a new task from an existing one
-pub fn make_task(func: anytype, args: anytype) !void {
-  const task = try task_alloc.create(os.thread.Task);
-  errdefer task_alloc.destroy(task);
-  try os.platform.new_task_call(task, func, args);
+pub fn wait() void {
+  os.platform.yield(false);
 }
 
+pub fn leave() noreturn {
+  os.platform.get_current_cpu().executable_tasks.leave();
+}
+
+const task_alloc = os.memory.vmm.backed(.Ephemeral);
+var balancer_lock = os.thread.Spinlock{};
+
+/// Allocating memory for the new task
 pub fn new_task() !*os.thread.Task {
   return task_alloc.create(os.thread.Task);
 }
 
+/// Creating a new task that calls a function call
```

Comment:
The comment should probably say you need to call this from an existing task

Reply:
Why? It is called from pretty much alive tasks, see AHCI

Reply:
Existing, not exiting